### PR TITLE
Implemented using aar consumer proguard configurations in ProguardMojo

### PR DESCRIPTION
**Pull request by hameno**
Gradle has a feature, where you can specify consumerProguardFiles to include
a proguard configuration into an aar artifact.
See https://plus.google.com/+IanLake/posts/1HfDuFFkMXG for more info about
consumerProguardFiles.

This pr only covers using the included proguard files. Including a proguard configuration into aars generated by maven is still open. 

refs #611

### CODE REVIEW

**Comment by hameno** [comment]
I'm not sure how to actually write an IT for this. I've tested it with my a project which uses play-services 7.0.0 and the generated temp_config.cfg includes the file from the play services (target\unpacked-libs\cgag_play-services-analytics_7.0.0\proguard.txt).

**Comment by hameno** [comment]
Implemented the second part (creation of the proguard.txt file)

File: src/main/java/com/jayway/maven/plugins/android/phase04processclasses/ProguardMojo.java [hunk 1]
```diff
@@ -429,7 +442,7 @@ private void executeProguard() throws MojoExecutionException
             IOUtils.write( commandStringBuilder, tempConfigFileOutputStream );
 
             executor.setCaptureStdOut( true );
-            commands.add( "@\"" + tempConfigFile.getAbsolutePath() + "\"" );
+            commands.add( "@" + tempConfigFile.getAbsolutePath() + "" );
```
Comment (mosabua) [inline src/main/java/com/jayway/maven/plugins/android/phase04processclasses/ProguardMojo.java]: Why did you remove the escaped " .. these are needed on Windows as far as I know.

Comment (hameno) [inline src/main/java/com/jayway/maven/plugins/android/phase04processclasses/ProguardMojo.java]: I had to remove this as it broke when the path has spaces inside. Only then it worked for me on windows.

Comment (mosabua) [inline src/main/java/com/jayway/maven/plugins/android/phase04processclasses/ProguardMojo.java]: I have a feeling this is breaking in some other situation then... Mac or Linux or something else. This was a pain to get right ..

Comment (Shusshu) [inline src/main/java/com/jayway/maven/plugins/android/phase04processclasses/ProguardMojo.java]: same feeling ;)

We'll see ^^

**Comment by william-ferguson-au** [comment]
Is there some documentation for this feature?

How do I define the config for my AAR such that it bundles the proguard config into the AAR and doesn't apply it during the AAR build?

Do I need to do anything in my APK config to ensure that it extracts and includes proguard config from dependent AARs?
